File: frontend/src/shared/api/modules/roles.ts
import { apiClient } from "../client";

export interface RoleItem {
  name: string;
  permissions: string[];
}

export interface RoleListResponse {
  items: RoleItem[];
  request_id?: string | null;
}

export async function fetchRoleList() {
  const { data } = await apiClient.get<RoleListResponse>("/roles");
  return data;
}

File: app/api/v1/schemas/roles.py
"""角色相关 Schema。"""

from __future__ import annotations

from pydantic import BaseModel, Field

from app.api.v1.schemas.common import RequestIdMixin


class RoleItem(BaseModel):
    """角色条目。"""

    name: str = Field(description="角色名称")
    permissions: list[str] = Field(description="权限列表")


class RoleListResponse(RequestIdMixin):
    """角色列表响应。"""

    items: list[RoleItem] = Field(description="角色列表")
